app/models/concerns/entity_events.rb

```diff
@@ -0,0 +1,35 @@
+module EntityEvents
+  extend ActiveSupport::Concern
+
+  included do
+    attr_accessor :event_tags
+
+    after_create do
+      data = entity_data(attributes)
+      send_event('entity_created', data) if data.any?
+    end
+
+    after_update do
+      data = entity_data(saved_changes)
```

Comment:
TVS have implemented this so that we always send all the attributes that have been requested, not just those that have changed ... and that's what I was expecting. Another one to check with @stevenleggdfe and maybe @veesaravi.

Reply:
@misaka , Its good practice to pull/send all attributes as we are doing ELT approach/design.. @stevenleggdfe good to know your view. Coming from Data modelling and Architect background.. good to get all data and then drive the data requirements and have change control in BQ (If needed)

Reply:
@misaka I agree with @veesaravi here. It's too common for us to be missing history, so it's useful to be able to rebuild the latest version of each entity from any given event (even a destroy/delete). It also speeds up analysis of updates - not sure what the BAT equivalent is but we do things like 'for applications from non-UK addresses, how long did it it take for people to finish typing in their qualifications' and so on. So please include all the data about an entity with each create/update/delete event, not just the things that changed.

Reply:
OK, great — that simplifies things slightly as well. Will amend.

Reply:
@duncanjbrown what would you think about having table_name as an event 'sub type' or something? That way, for other events that aren't entity in future, we could use the field for those as well. It'd give us some wriggle room to store extra stuff without putting it into the key/value struct.

Reply:
25de7e8